extensions/buildpack-registry.md

```diff
@@ -0,0 +1,118 @@
+# Buildpack Registry
+
+The Buildpack Registry supports distribution of buildpacks. It provides a centralized service that platforms can use to resolve a buildpack ID and version into a concrete buildpackage that can be downloaded and used.
+
+## Buildpack ID
+
+Each buildpack in the registry MUST have an ID that adheres to the following restrictions:
+
+* It MUST contain a single `/` character
+* The value to the left of the `/`, called the namespace, MUST match `[a-z0-9\-\.]{1,253}`
+* The value to the right of the `/`, called the name, MUST match `[a-z0-9\-\.]{1,253}`
+
+## Backing Storage
+
+The registry information is stored in a repository called an index. The index is a set of files and directories that contain information about available buildpacks.
+
+The registry index MUST be stored in a Git repository.
+
+The index MUST work across all major Operating Systems.
+
+### Directory Structure
+
+Folders in the index MUST be split by two nested folders. The buildpack ID will determine the name of the folders where:
+* The first folder MUST represent the first two characters
+* The second folder MUST represent the third and fourth characters
+* IDs that are 1-3 characters long MUST be represented by special folders.
+
+The folders will contain files that represent each buildpack ID where the following is true:
+* The `/` character in the ID MUST be replaced with `_`
+* The namespace MUST match `[a-z0-9\-\.]{1,253}`
+* The name MUST match `[a-z0-9\-\.]{1,253}`
+
+Here's an example directory structure:
+
+```
+1
+├── example_a
+└── example_b
+2
+├── example_aa
+└── example_ab
+3
+├── a
+│   ├── foobar_abc
+│   └── example_acd
+└── b
+    ├── example_bcd
+    └── example_bed
+fo
+├── ob
+│   ├── samples_fooball
+│   └── example_foobar
+└── oc
+    └── example_foocal
+```
+
+The following ids are reserved by Windows, so they aren't allowed as valid ids:
```

Comment:
Probably should have asked about this on the RFC, but I don't understand this part.

Reply:
@ekcasey I'm trying to remember my discussion with Sean from the crates.io team, but these names are reserved by windows as filenames.

Reply:
From the last leadership meeting: Due to the `_` after 2 characters, these names will never come up. Let's cut this section.

Reply:
@jkutner do you want to make this change?

Reply:
done